src/mono/mono/utils/mono-time.h

```diff
@@ -32,7 +32,7 @@ gint64 mono_100ns_datetime_from_timeval (struct timeval tv);
 #if defined(HOST_DARWIN)
 #include <mach/clock.h>
 typedef clock_serv_t mono_clock_id_t;
-#elif defined(__linux__)
+#elif defined(__unix__)
 #include <sys/types.h>
 typedef clockid_t mono_clock_id_t;
```

Comment:
Can we use a configure test to look for `clockid_t` in `sys/types.h` instead of guessing based on OS?

Reply:
Somethign like how we check for `blksize_t`:

```
	AC_CHECK_TYPES([blksize_t], [AC_DEFINE(HAVE_BLKSIZE_T)], , 
		[#include <sys/types.h>
		 #include <sys/stat.h>
		 #include <unistd.h>])
```

Reply:
@rootwyrm would this work for FreeBSD?

Reply:
yes

Reply:
Beaten to the punch! Which is a-okay by me, as I did need to noodle a bit. This should be fine for AIX and PASE as well.

Reply:
I think that it is safe to just check the existence of "sys/types.h".

Reply:
Why not just rely on `time.h` which is where POSIX defines `clock_getres`, `clock_gettime`, and `clock_settime` and therefore must also define or import the file which includes `clockid_t`?

Reply:
For reference: https://pubs.opengroup.org/onlinepubs/9699919799/basedefs/time.h.html

> [CX] The <time.h> header shall define the clockid_t and timer_t types as described in <sys/types.h>.

Reply:
make sense

Reply:
I may be misreading things, but it would appear to me [MUSL time.h](https://git.musl-libc.org/cgit/musl/tree/include/time.h?id=0b44a0315b47dd8eced9f3b7f31580cf14bbfc01) might have problems with that method.

Reply:
I believe that is the point of `#define __NEED_clockid_t` and then `#include <bits/alltypes.h>`

So `time.h` still "defines" `clockid_t`, just indirectly.